ci/terraform/oidc/spot-sqs.tf

```diff
@@ -27,6 +27,48 @@ resource "aws_sqs_queue" "spot_request_dead_letter_queue" {
   tags = local.default_tags
 }
 
+data "aws_iam_policy_document" "cross_account_spot_request_queue_policy_document" {
+  statement {
+    sid    = "AllowSpotAccountToReceive"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${aws_ssm_parameter.spot_account_number.value}:root"]
+    }
```

Comment:
I think the identifier can be shortened to just `[${aws_ssm_parameter.spot_account_number.value}]` ([example here](https://github.com/govuk-one-login/authentication-api/pull/4250/files#diff-331c55b2d84d1db997c377c4169580d4c13c837f9193e2217a8814089ed948e9))

Reply:
Oh yeah, thanks!